### DISCUSSION TITLE
The Next Generation of Genealogy Sitebuilding (TNG)

### DISCUSSION THREAD

**Discussion by jpjoneshy1** [e107inc/e107#5354] [Q&A Plugins]
Hello and new to this discussion forum.

I used to use e107 several years back in it's v.7 days.  We (the TNG users) had an e107 plugin used to integrate e107 and TNG, but like I said was back in the e017 v .7 and TNG v. 6-7 days.  

I still have a copy of the e107/TNG plugin and was wondering if the are any e107 plugin developers who would be willing to update the plugin I have to work with the current TNG and php versions.  I'm willing to pay $$ for it (depending on how much $$).  And since TNG is a paid application,  I can set it up for you on a live site and give admin access.  

I would need to know what/how to set it all up, like install e107 in the root and create a tng folder under that, etc....

If anyone is interested in this project, please contact me at jpjoneshy1 at gmail dot com

Thanks,
John Paul

**Comment by PH5HP** [comment]
Hi,

No answers (yet). I like E107 as my CMS, but it is a little disappointing that a lot of older plugins simply not work with the latest version of E107 despite that they are downloadable in the plugin section. I think that this is for a lot of people a reason to abandon E107.

They had to consider backwards compatability, but they did not. This was not a problem if they made all plugins compatible with the current version. I wonder if you ever get an answer on your question above.

Kind regards,
Henk - ph5hp.nl

**Reply by Jimmi08** [reply]
I respect your point, but I can't entirely agree with it. The problem is that those plugins are downloadable at all.  Those plugins will not work with the newest PHP version, most of their functionality is already in core, and there is an issue of safety too, bad written code etc.  And a lot of plugins are not in the official repo.     

The core itself without some plugins is pretty cool but you need to know how to use it.  Times are changing. There is no community to give you support if you want to continue.  It is a general problem for any CMS nowadays.  Not to mention that people are older, have different priorities, value free time, or need work for money, they have families...  

If anybody wants to use e107, they should learn the basics of coding and do it yourself.  

By the way, backward compatibility is a nightmare that only complicates new development.

**Reply by PH5HP** [reply]
Hi Jimmi08,

I respect all the points you have. 

But why are those plugins downloadable at the E107 website and in E107 itself if they do not work in that version anymore? Do they not know if a plugin works in v2 of E107 or not? I ended up locking up my E107 website after installing a plugin which did not work.

Where is the official repo? I thought the plugins of the E107 were in the official repo? I would be a lot better if all was clear which plugins are working in E107 V2 and which are not.

I know that things are changing all the time and people get other priorities, but maybe we could adjust/rewrite some of the plugins which are not in the core of E107 and make some of them working again and. One at the time.

Not everyone can code. For myself I am not good enought in PHP coding to do the things I want to do. Still learning about PHP every day. I made a plugin for my HamRadio hobby and it even works running E107 V2, but I do not know if it meets the requirements of safety of E107 V2. The problem is that a lot of knowledge is not there or does not have examples for people like me.

Anyway. Thanks for answering my post here. 

Kind regards,
Henk - ph5hp.nl

**Reply by Jimmi08** [reply]
`the official repo `- it is e107.org site, my wrong wording.  I don't have answers to your questions related to it.  

`Not everyone can code. For myself I am not good enough in PHP coding to do the things I want to do. Still learning about PHP every day. I made a plugin for my HamRadio hobby and it even works running E107 V2, but I do not know if it meets the requirements of safety of E107 V2. The problem is that a lot of knowledge is not there or does not have examples for people like me.`

I am in the same position as you.  Now with GPT it is ten times easier than before.  But still... 

I had a safety incident 1-2 ago on the most important site that I am running for 15 years. If you check gitter, for 24 hours nobody was able to help me.  People have been asking how to block in v2 IP's range for years because asterix is not working.  I had to learn about CIDR (Classless Inter-Domain Routing)  myself to fix this.  Then I created file that would be adding IP trying to use this URL to banlist - f.e. wp-login.php if it is not WP site, etc. but IP wasn't blocked. Was it because localhost testing?  5-minute answer from the developer instead of my 5 hours digging in code.  What to say more? It is so frustrating. 

By the way, that problem was caused old Nuke code in the module and thanks e107 I was able to detect it, solve and get that IP range blocked on my hosting provider level. 

I need to go now, but I am still here and  I am using e107 for amazing things.  I just don't have time to write about them.

**Comment by Vodhin** [comment]
Hi. I have just looked around for [The Next Generation](https://www.tngsitebuilding.com/) and it looks like you really don't need e107 to run this software. In fact, it is probably better to install it as a stand-alone solution. If your website is more than just a Genealogy site, eg. you also have other sections for gaming, movie reviews, etcetera, then you should be able to install the TNG software into a subfolder of your website and simply add links on pages from which you want people to find it. You could also install it as a subdomain of your main site, which might be more pleasing when sharing the URL: my http://www.vodhin.org also has a subdomain https://estate.vodhin.org which operates independently.

**Reply by PH5HP** [reply]
Hi,

Yes, that is true. The Next Generation is a php-based website packadge for Genealogy. I have one running for my own family tree at https://pit-dehaan.nl/   (if you are interested to see what it is).

On the other hand I think that maybe the plugin has some other display functionality to embed information of someones family tree in a E107 website. I do not know, because I did not know about the existance of such a plugin for the past versions of E107. Maybe someone else can shine a light on that?

I am interested in the plugin itself. Maybe we can make it function again in V2 of E107. Anyone who can provide me with the source of The Next Generation Genealogy plugin?

Kind regards,

Henk - PH5HP - https://ph5hp.nl/

**Reply by Vodhin** [reply]
On the e107 Plugins page I found [TNG Integrated Login v1.2](https://e107.org/plugins/?iframe=1&id=637) . It looks like what you wanted. I've uploaded the plugin to my [Estate Demo website](https://estate.vodhin.org) and have started digging around. The code is well out of standards (both e107 and general PHP scripting) and may have security issues. It also has some basic errors, including a couple of admin links to files that are not included in the plugin download and missing/incorrect Language Constants (hard coded text used in forms). Give me a week or so to see if I can get the PHP files up to snuff and then we can look at connecting to a database (I might be able to make a dummy DB for testing).

**Reply by PH5HP** [reply]
Hi,

Indeed.. It looks like that this was what I wanted to look at. I see that it is ment to work with version 6 of TNG. That is a long time ago. At the moment we are using 14.0.5 version of TNG. I also will take look what it is all about, but I am afraid my knowledge of PHP is limited.

Thanks for the link for downloading this plugin.

Kind regards,
Henk - ph5hp.nl

**Reply by Vodhin** [reply]
I haven't forgotten about this. I've been fixing bugs in my own work. I would need to know the database structure for the table that holds users in the current version of TNG The easiest way is to right click on a user's form and look at all the input fields' NAMES eg
`<input type="text" name="some_name" .../>`

**Reply by PH5HP** [reply]
Hi,

Do you know what the purpose of the TNG plugin was? Just displaying some data?

**Comment by tgtje** [comment]
@PH5HP a login way for  integrated login between e107 v0.7+ and The Next Generation > genealogy 

no ffence please but best to ask in separate issue or request or Gitter. ;-) (to prevent contamination of a particular issue).

**Reply by PH5HP** [reply]
HI tgtje,

Just was curious, but I have no plans using it myself as I have bought TNG myself as a website.

**Reply by tgtje** [reply]
@PH5HP i understand,, no problem. 
Of course info is sometimes hard to find, 
Reading the topic, and as @jpjoneshy1 mentioned...  ( + @Vodhin) maybe you should make contact and exchange things..
Since both are familiar and/or using TNG.. work together?. Might benefit both ( and future others? who knows..).
Any outcome of such would be benificial and should become a topic on its own, similar to the Real Estate discussion. (imo the best way, as others would be handed a way to paticipate (iff interested).
I think the best answers to both your questions and experiences can be shared, as other responses are scarse in regards.
Just be bold enough..

**Reply by Vodhin** [reply]
I am fine keeping all questions/comments regarding this plugin here in this thread. When I get closer to testing I’ll create a repository and link it here. 

What this plugin did do is automatically add e107 users to the TNG database and also provided a couple of menus to integrate some TNG features on e107 generated pages

**Reply by Vodhin** [reply]
@PH5HP - something in one of your earlier posts had been gnawing at the back of my mind, so I had to go and have reread:

`For myself I am not good enough in PHP coding to do the things I want to do... I made a plugin for my Ham Radio hobby...`

You've peaked my interest. My dad was originally W2GHR (1950's) and then W1JTB (1960's - "W1 Just Too Bad" and still active as a repeater in Graniteville, VT) and W2NH (1990's until he went silent key in 2022). I'm not a HAM myself (my many interests take up all my time and brain power) but If you are still using that plugin and would like someone to make it fresh for today's mobile world, perhaps I could take a look at it? If you are interested, you can attach a zip file  of the plugin here, and the current e107 has an easy way to export its data (if any). Let me k now.

**Comment by Vodhin** [comment]
Here is a test [https://estate.vodhin.org/tngIL2.zip](https://estate.vodhin.org/tngIL2.zip)
This is a fresh build based on information discovered in the original plugin. If anyone with TNG installed wants to download and test it, please be advised that the only working section is the admin area form, specifically the DB configuration. What I need to know - at the moment - is if you can get the TNG users table ("tng_users" was the old default) connected: if you do, then the Test Results Tab will display a list of the field names in the TNG Users table. Copy and paste that list here so I can confirm that table's structure. Thank you.

**Comment by Vodhin** [comment]
Development of a new version is underway. The old plugin repository is incomplete (there is a menu link to `/admin/index.php` but the entire admin folder is missing) and I do not know the current TNG database structure. So this new version should allow you to connect to your TNG database by filling in the DB Configuration and if it successfully connects the Test Results tab will list the tables and their structure (I have tested this function with a dummy database and it seems to work). 

Please download  [https://estate.vodhin.org/tngIL2.zip](https://estate.vodhin.org/tngIL2.zip) and copy/paste the table structure so I can make sure the new version connects with the proper fields.

**Comment by Vodhin** [comment]
@jpjoneshy1 @PH5HP Is there still any interest in this? To continue I need to know the TNG database structure. If you or anyone with TNG running somewhere is interested, the new TNG plugin https://estate.vodhin.org/tngIL2.zip will output that table structure which can be copied and pasted here like so:

setup database connections in the Configuration Tab:
![dbres01](https://github.com/user-attachments/assets/f41d7b3b-186f-47ba-956f-55f2b1370f35)


Copy and paste each table in the Test Results Tab: 
![dbres02](https://github.com/user-attachments/assets/ca1c3801-3d6a-4de3-b6ff-8bfe1ab42543)


Result:

<html><body>
<div class="admin-container container-fluid"><div class="row is-table-row "><div class="admin-right-panel col-md-9 col-lg-10"><div class="admin-main-content"><div class="block"><div class="block-text"><div class="tab-content tngClassTabs"><div class="tab-pane active" id="tab-1" role="tabpanel"><div class="tngTestDiv1"><div class="tngTestDiv2">
            <h4>tng_users</h4>
            
Field | Type | Null | Key | Default | Extra
-- | -- | -- | -- | -- | --
description | text | NO |   |   |  
username | varchar(55) | NO | PRI |   |  
password | varchar(40) | NO |   |   |  
gedcom | varchar(50) | NO |   |   |  
allow_edit | tinyint(1) | NO |   |   |  
allow_add | tinyint(1) | NO |   |   |  
tentative_edit | tinyint(1) | NO |   |   |  
allow_delete | tinyint(1) | NO |   |   |  
allow_lds | tinyint(1) | NO |   |   |  
allow_ged | tinyint(1) | NO |   |   |  
allow_living | tinyint(4) | NO |   |   |  
branch | varchar(55) | NO |   |   |  
realname | varchar(55) | NO |   |   |  
email | varchar(55) | NO |   |   |  
notes | text | NO |   |   |  
</body>
</html>

**Reply by PH5HP** [reply]
Hi,

I can't get it working here. I filled all the fields like I should, but it will not connect to the database. Here a screenshot of the structure of the users table.

Kind regards,
Henk - ph5hp.nl

![image](https://github.com/user-attachments/assets/bd6023ba-4223-400a-82c0-86fc61df8f18)

**Reply by Vodhin** [reply]
Thank you for posting that screenshot of the TNG user table. It tells me that a lot has changed since this old plugin was created. It may not be worth the effort to rebuild it. It might be better to create a new genealogy Plugin for e107, though I would have to figure out a suitable DB structure, preferably compatible with TNG (assuming that you can export your TNG data). 

I am curious why it won't connect for you. Check to see if `tngconfig.php` in the tngIL2 folder contains the correct information. It should look something like this:
```
<?php
$this->tng_dbhost = "localhost";
$this->tng_dbname = "dbPrefix_dbName";
$this->tng_dbuser = "dbPrefix_dbUserName";
$this->tng_dbpass = "yourDBPassword";
$this->tng_usertab = "tng_users";
?>
```

If it does NOT have your configuration make sure the CHMOD for `tngconfig.php` is set to 644.
If it does have your configuration AND `$this->tng_dbhost = "localhost";` then I am not sure why it will not connect.
If `$this->tng_dbhost = "some/other/path"` then I suspect that either the path is wrong or the server's security settings are preventing cross-origin connections.

**Reply by PH5HP** [reply]
HI,

Connection established (success). It was a matter of directory name of the plugin on the server which prevented me to connect with the TNG database.  However I can't see any tables in the database when pressing the TEST RESULTS tab.

Kind regards,
Henk - PH5HP.nl

![image](https://github.com/user-attachments/assets/b6d8dc0f-0985-4471-bbf4-bca88b7c58e5)

![image](https://github.com/user-attachments/assets/29e91cc5-77ef-414c-a2dd-9f84c1aea2da)

**Reply by tgtje** [reply]
@Vodhin @PH5HP just as sidenote... to be considered ??    from image posted there's **utf8mb3** 

The utf8mb3 character set is deprecated. utf8mb3 remains supported for the lifetimes of the MySQL 8.0.x and MySQL 8.4.x LTS release series.

**Reply by Vodhin** [reply]
That’s good it connected. Not so good that no tables were read. I was usuing e107’s SQL functions to discover and read the tables and it looks like the TNG database is using another format. 

For now remove the plugin and delete its files: I realize that there is no protection for the tngconnect.php file, which can reveal your db configuration.

**Comment by PH5HP** [comment]
Hi,

Did you find this?

https://tng.community/index.php?/forums/forum/34-e107/

**Reply by tgtje** [reply]
wow not too busy there. i think @PH5HP refers to https://tng.community/index.php?/forums/topic/5345-planning-for-tngil-v20/
the last post. who knows ?
